### PR TITLE
feat: Add JP version and regex operations in jsonlogic

### PR DESCRIPTION
**Pull request by ayushjain17**
## Problem
Semantic versioning and regex related operations not present in jsonlogic

## Solution
Add semver and regex operations in jsonlogic

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -41,7 +41,7 @@ chrono = { version = "0.4.26", features = ["serde"] }
 uuid = {version = "1.3.4", features = ["v4", "serde"]}
 reqwest = { version = "0.11.18", features = ["json"]}
 jsonschema = "~0.17"
-jsonlogic = "0.5.1"
+jsonlogic = { git = "https://github.com/juspay/jsonlogic_rs.git", rev = "4f1a8bcbb423c599eacf6c95ae17f5c0974bc071" }
```
Comment (Datron) [inline Cargo.toml:44]: Why did we make this change? All other changes are in JS

File: clients/js/src/index.ts [hunk 2]
```diff
@@ -15,7 +15,12 @@ export class CacReader {
     defaultConfig: IObject;
 
     static {
-        jsonLogic.add_operation(">>", compareSemanticIsGreater);
+        jsonLogic.add_operation("match", matchRegex);
+        jsonLogic.add_operation("semver_eq", compareSemanticIsEqual);
+        jsonLogic.add_operation("semver_gt", compareSemanticIsGreater);
+        jsonLogic.add_operation("semver_gteq", compareSemanticIsGreaterEqual);
```
Comment (Datron) [inline clients/js/src/index.ts:21]:
```suggestion
        jsonLogic.add_operation("semver_ge", compareSemanticIsGreaterEqual);
```

File: clients/js/src/index.ts [hunk 3]
```diff
@@ -15,7 +15,12 @@ export class CacReader {
...
+        jsonLogic.add_operation("semver_lt", compareSemanticIsLesser);
+        jsonLogic.add_operation("semver_lteq", compareSemanticIsLesserEqual);
```
Comment (Datron) [inline clients/js/src/index.ts:23]:
```suggestion
        jsonLogic.add_operation("semver_le", compareSemanticIsLesserEqual);
```

File: clients/js/src/utils/operations.ts [hunk 4]
```diff
@@ -1,9 +1,33 @@
-const compareSemanticIsGreaterImp = function (version_a: string, version_b: string) {
-    return version_a.localeCompare(version_b, undefined, {numeric:true, sensitivity:'base'}) > 0;
+const compareSemantic = function (version_a: string, version_b: string): number {
+    return version_a.localeCompare(version_b, undefined, { numeric: true, sensitivity: 'base' });
```
Comment (Datron) [inline clients/js/src/utils/operations.ts:2]: the `locales` parameter and `options` parameter [are not supported](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#browser_compatibility) on android webview
Screenshot 2024-05-24 at 12 21 32 PM

**Review by Datron — CHANGES_REQUESTED** [review]
Lack of support for android webviews means this won't work with SDK. Is that fine?

File: clients/js/src/index.ts (hunk 2 above)
Comment (pratikmishra356) [inline clients/js/src/index.ts:21]: @ayushjain17 should we add one for between also ?

Comment (ayushjain17) [reply clients/js/src/index.ts]: @pratikmishra356 less than supports between operation as well, the behaviour of these version comparison operators are same as that of the normal comparison operators - wrt the number of operators

File: Cargo.toml (hunk 1 above)
Comment (ayushjain17) [reply Cargo.toml]: to add the same operators in the rust client

Comment (Datron) [reply Cargo.toml]: Are those added in this PR? If not, can we make this change when we add the operations

Comment (ayushjain17) [reply Cargo.toml]: https://github.com/juspay/jsonlogic_rs/pull/1

**Review by ShubhranshuSanjeev — APPROVED** [review]

File: Cargo.toml [hunk 5]
```diff
@@ -41,7 +41,7 @@ chrono = { version = "0.4.26", features = ["serde"] }
 uuid = {version = "1.3.4", features = ["v4", "serde"]}
 reqwest = { version = "0.11.18", features = ["json"]}
 jsonschema = "~0.17"
-jsonlogic = "0.5.1"
+jsonlogic = { git = "https://github.com/juspay/jsonlogic_rs.git", version = "0.5.2" }
```
Comment (Datron) [inline Cargo.toml:44]: Why add this without rust changes?

**Review by Datron — APPROVED** [review]

**Pull request merged**